### ISSUE TITLE
Organizing credentials and notes in folders

### ISSUE THREAD

**Issue by waltertamboer** [ownpass/ownpass#5]
When dealing with big amount of credentials and notes, it would be great if they can be organized in a folder structure. This way credentials and notes can be grouped in categories (the folders).

**Comment by ei8fdb** [comment]
### Organising passwords into folders

I currently store my passwords in various folders inside my password manager. And it confuses the hell out of me! 

I always come up with weird ways to save my passwords, and I end up forgetting why I put _that_ password in *there. :)

A first step might be to give the user a powerful, and fast search facility. Design wise it would be very simple. It might be difficult from a development point of view?

If you like I can sketch something up to explain how it could work?

Organising passwords into folders would be useful. Once the goal of the software "to manage and protect passwords" was successful.

### Notes

I see "notes" as an added, secondary feature - not central to the main function of a password manager.

Yes nice to have, but after the main purpose of the password manager has been achieved.

**Comment by waltertamboer** [comment]
Good point, search might even be more important than organizing into folders (?)

> It might be difficult from a development point of view?

It depends on the things we search on. Currently we don't have that much data available.
- Name
- Description
- URL of login page (splitted into scheme, host, path, etc.)
- Identity

It would be fairly simple to just search in these columns for matching results. It's not like we need an extensive searching functionality. At least, I guess?

> If you like I can sketch something up to explain how it could work?

The more sketches, the better :+1: :)

> I see "notes" as an added, secondary feature - not central to the main function of a password manager.

I agree but we should keep in mind that one maybe wants to save passwords not related to websites directly. For example, the WIFI password, or the password of a safe.
